data/scripts/set_wallpaper

```diff
@@ -128,16 +128,22 @@ fi
 if [ "${KDE_FULL_SESSION}" == "true" ]; then
     # Plasma 5.7 introduced a new feature to set the wallpaper via a dbus script:
     # https://github.com/KDE/plasma-workspace/commit/903cbfd7e267a4812a6ec222eb7e1b5dd775686f
-    if [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" == '5' ]]; then
-        dbus-send --type=method_call --dest=org.kde.plasmashell /PlasmaShell org.kde.PlasmaShell.evaluateScript string:"
+    plasma_qdbus_script="
         let allDesktops = desktops();
         for (let d of allDesktops) {
             if (d.wallpaperPlugin == 'org.kde.image') {
                 d.currentConfigGroup = Array('Wallpaper', 'org.kde.image', 'General');
-                d.writeConfig('Image', 'file://""$WP""')
+                d.writeConfig('Image', 'file://""$WP""');
             }
         }
-        "
+    "
+    if [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" == '6' ]]; then
+        dbus-send --type=method_call --dest=org.kde.plasmashell /PlasmaShell org.kde.PlasmaShell.evaluateScript string:"$plasma_qdbus_script"
+        # Reuse the exit code from dbus
+        exit "$?"
+
+    elif [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" == '5' ]]; then
```

Comment:
Can we change this to check KDE_SESSION_VERSION >= 5 instead? It would be better to not define the same dbus-send call in two separate branches.

(I think we can remove KDE 4 support at this point too, since I don't think modern distros ship it anymore)

Reply:
The reason I have two different branches is because the "locked widgets" warning does not make sense in Plasma 6 - it doesn't make sense for Plasma 5 since about 2020 either, but I wasn't sure about removing it, though I think it probably can be removed.

I can also refactor the code the use `>= 5` and have the "locked widgets" error only if the version is 5, if you prefer.

Regarding the `else` clause under the KDE, it may still be useful, though weirdly it doesn't exit after populating the `variety-wallpaper` directory, so the code continues and runs `gsettings set org.gnome.desktop.background` - which is a weird choice...

Reply:
> I can also refactor the code the use >= 5 and have the "locked widgets" error only if the version is 5, if you prefer.

Yes, this is exactly what I mean

Reply:
done.